DriverLoader.cpp

```diff
@@ -91,12 +93,19 @@ namespace vive {
             throw CouldNotLoadEntryPoint();
         }
         factory_ = reinterpret_cast<DriverFactory>(proc);
-#elif defined(OSVR_MACOSX)
-#error                                                                         \
-    "Implementation incomplete! Load dynamic library here and retrieve entry point function here!"
-#elif defined(OSVR_LINUX)
-#error                                                                         \
-    "Implementation incomplete! Load dynamic library here and retrieve entry point function here!"
+#elif defined(OSVR_LINUX) || defined(OSVR_MACOSX)
+        impl_->driver_ = dlopen(driverFile.c_str());
+        if (!impl_->driver_) {
+            reset();
+            throw CouldNotLoadDriverModule(dlerror());
+        }
+
+        auto proc = dlsym(impl_->driver_, ENTRY_POINT_FUNCTION_NAME);
```

Comment:
dlerror() returns NULL when is no error. But in that case, this exception will be never thrown.

Also, calling dlerror() without taking returned pointer to string -- is really bad idea.
POSIX's libdl gives really nice reports through dlerror(). For example, in case loading 64 bit library in 32 bit app gives "Wrong elf class ELFCLASS64". So user can know where's the problem. And developer, ofc. :)